src/tof/model.py

```diff
@@ -2,16 +2,36 @@
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
 from itertools import chain
-from typing import List, Union
+from typing import Dict, List, Optional, Tuple, Union
 
-import matplotlib.pyplot as plt
-import numpy as np
 import scipp as sc
-from matplotlib.collections import LineCollection
 
 from .chopper import Chopper
 from .detector import Detector
 from .pulse import Pulse
+from .result import Result
+
+ComponentType = Union[Chopper, Detector]
+
+
+def _input_to_dict(
+    obj: Optional[
+        Union[
+            Dict[str, ComponentType],
+            List[ComponentType],
+            Tuple[ComponentType, ...],
+            ComponentType,
+        ]
+    ]
+):
```

Comment:
Would this have to have a default of `None` to be optional? In any case,
```suggestion
def _input_to_dict(
    obj: Union[
            None,
            Dict[str, ComponentType],
            List[ComponentType],
            Tuple[ComponentType, ...],
            ComponentType,
        ]
):
```
would be more readable.